[release/2.11] Generate changelog for 2.11.5

sergepetrenko opened the pull request:
Also, remove unreleased/ entries.

NO_DOC=changelog
NO_TEST=changelog
NO_CHANGELOG=changelog

Totktonada commented on changelogs/2.11.5.md:
```diff
@@ -0,0 +1,149 @@
+* Fixed a bug when an SQL count statement wasn't tracked by MVCC if it was
+  the first in a transaction (gh-10825).
+
+#### Datetime
+
+* Forbid non-integers in SQL's `CAST({}) AS datetime` (gh-10391).
```
Nit: I don't think that a single item deserves its one fourth level section. I would include it to `SQL`.

Totktonada reviewed (approved):
I think we've no goal to re-verify the tool for generation these files, so I only glanced over the items briefly.

I see no flaws, LGTM.

If a more thorough review is needed, let me know.

xuniq commented on changelogs/2.11.5.md:
```diff
@@ -0,0 +1,149 @@
+* Fixed a memory leak on foreign key constraint check failure (gh-10476).
+* Fixed a memory leak on an attempt to create or modify a user denied due to
+  access restrictions (gh-10485).
+* Fixed a crash when `slab_alloc_factor` is low and memory pressure is high
+  (gh-10148).
+* Fixed a crash when the first of two consequent DDL operations gets
+  rolled back due to WAL failure (gh-10235).
```
```suggestion
* Fixed a crash when `slab_alloc_factor` was low and memory pressure was high
  (gh-10148).
* Fixed a crash when the first of two consequent DDL operations was
  rolled back due to WAL failure (gh-10235).
```

xuniq commented on changelogs/2.11.5.md:
```diff
@@ -0,0 +1,149 @@
+* Fixed a bug when `index:count()` could return a wrong number, raise the
+  last error, or fail with the `IllegalParams` error if the index has
+  the `exclude_null` attribute and MVCC is enabled (gh-10396).
+* Fixed a crash on transaction rollback if memory usage is close to the limit
```
```suggestion
* Fixed a crash on transaction rollback if memory usage was close to the limit
```

xuniq commented on changelogs/2.11.5.md:
```diff
@@ -0,0 +1,149 @@
+* Fixed a crash when using pagination over a non-unique index with range
+  requests and MVCC enabled (gh-10448).
+* Fixed a bug when `index:count()` could return a wrong number, raise the
+  last error, or fail with the `IllegalParams` error if the index has
+  the `exclude_null` attribute and MVCC is enabled (gh-10396).
```
```suggestion
  last error, or fail with the `IllegalParams` error if the index had
  the `exclude_null` attribute and MVCC was enabled (gh-10396).
```

xuniq commented on changelogs/2.11.5.md:
```diff
@@ -0,0 +1,149 @@
+* Dropped finalizer table rehashing after GC cycle (gh-10290).
+* Fixed recording of `select(string, ...)`.
+* Fixed stack allocation after on-trace stack check.
+* Fixed recording of `__concat` metamethod that throws an error.
```
```suggestion
* Fixed recording of the `__concat` metamethod that threw an error.
```

xuniq commented on changelogs/2.11.5.md:
```diff
@@ -0,0 +1,149 @@
+
+### Upgrade
+
+* Automatically fix incorrect format of user-defined spaces and empty password
+  for users during upgrade from old schema versions (~1.6.8) (gh-10180).
```
```suggestion
* The incorrect format of user-defined spaces and empty password
  for users during upgrade from old schema versions (~1.6.8) are fixed automatically now (gh-10180).
```

Totktonada commented on changelogs/2.11.5.md:
```diff
@@ -0,0 +1,149 @@
+# 2.11.5
+
+Date: 2024-11-21
```
I guess, at least 2024-11-22 (2024-11-21 is valid for 20 next minutes, though).

Totktonada commented on changelogs/2.11.5.md:
Should we also update something in `src/box/lua/upgrade.lua`?

I just remember that it was a problem in some past releases.

sergepetrenko replied:
This time `downgrade versions` were already updated accordingly before:
https://github.com/tarantool/tarantool/blob/1b12f2417ba213b1fed7015eed6a517deb3fe9f5/src/box/lua/upgrade.lua#L2093-L2100

sergepetrenko commented on changelogs/2.11.5.md:
```diff
@@ -0,0 +1,149 @@
+  MVCC enabled that could lead to unexplainable errors (gh-10262).
+* Fixed a bug when the `drop_while` method of index iterators (`index:pairs()`,
+  `space:pairs()`) dropped an extra element (gh-6403).
+
+
```
Thanks. That's an oversight after some corrections.

sergepetrenko commented:
@Totktonada @xuniq thanks for your reviews!
<details><summary>Here's the diff:</summary>

```diff
diff --git a/changelogs/2.11.5.md b/changelogs/2.11.5.md
index fe9482b363..21f43b186d 100644
--- a/changelogs/2.11.5.md
+++ b/changelogs/2.11.5.md
@@ -1,12 +1,12 @@
 # 2.11.5
 
-Date: 2024-11-21
+Date: 2024-11-22
 Tag: 2.11.5
 
 ## Overview
 
 2.11.5 is the next stable release in the [long-term support (LTS)
-version][release_policy][release_policy] 2.11.x release series.
+version][release_policy] 2.11.x release series.
 
 The label "stable" means there are 2.11.x-based applications running in
 production for quite a while without known crashes, incorrect results or
@@ -20,7 +20,7 @@ This release resolves 52 issues since the 2.11.4 version.
 ## Compatibility
 
 Tarantool 2.11.x is backward compatible with Tarantool 1.10.x in binary data
-layout, client-server protocol and replication protocol.
+layout, client-server protocol, and replication protocol.
 
 Please [upgrade][upgrade] using the `box.schema.upgrade()` procedure to unlock
 all the new features of the 2.x series.
@@ -35,24 +35,23 @@ all the new features of the 2.x series.
 * Fixed a memory leak on foreign key constraint check failure (gh-10476).
 * Fixed a memory leak on an attempt to create or modify a user denied due to
   access restrictions (gh-10485).
-* Fixed a crash when `slab_alloc_factor` is low and memory pressure is high
+* Fixed a crash when `slab_alloc_factor` was low and memory pressure was high
   (gh-10148).
-* Fixed a crash when the first of two consequent DDL operations gets
+* Fixed a crash when the first of two consequent DDL operations was
   rolled back due to WAL failure (gh-10235).
 * Fixed an inconsistency between internal caches and system spaces with
   MVCC enabled that could lead to unexplainable errors (gh-10262).
 * Fixed a bug when the `drop_while` method of index iterators (`index:pairs()`,
   `space:pairs()`) dropped an extra element (gh-6403).
 
-
 ### Memtx
 
 * Fixed a crash when using pagination over a non-unique index with range
   requests and MVCC enabled (gh-10448).
 * Fixed a bug when `index:count()` could return a wrong number, raise the
-  last error, or fail with the `IllegalParams` error if the index has
-  the `exclude_null` attribute and MVCC is enabled (gh-10396).
-* Fixed a crash on transaction rollback if memory usage is close to the limit
+  last error, or fail with the `IllegalParams` error if the index had
+  the `exclude_null` attribute and MVCC was enabled (gh-10396).
+* Fixed a crash on transaction rollback if memory usage was close to the limit
   (gh-10551).
 * Fixed several bugs when DDL with MVCC enabled could lead to a crash
   or violate isolation of other transactions (gh-10146).
@@ -91,7 +90,6 @@ all the new features of the 2.x series.
 * Fixed a memory leak on disconnection from replica (gh-10480).
 * Optimized synchronous transaction queue processing (gh-9917).
 
-
 ### LuaJIT
 
 Backported patches from the vanilla LuaJIT trunk (gh-10199). The following
@@ -105,7 +103,7 @@ issues were fixed as part of this activity:
 * Dropped finalizer table rehashing after GC cycle (gh-10290).
 * Fixed recording of `select(string, ...)`.
 * Fixed stack allocation after on-trace stack check.
-* Fixed recording of `__concat` metamethod that throws an error.
+* Fixed recording of the `__concat` metamethod that threw an error.
 * Fixed bit op coercion in `DUALNUM` builds.
 * Fixed 64-bit shift fold rules.
 * Fixed loop optimizations for cdata arguments of vararg FFI functions.
@@ -122,7 +120,7 @@ issues were fixed as part of this activity:
 
 #### Datetime
 
-* Fixed a bug that caused `datetime.parse()` ignore `tzoffset`
+* Fixed a bug that caused `datetime.parse()` ignore the `tzoffset`
   option if a custom format was used (gh-8333).
 * Forbid non-integers in `datetime.new()` (gh-10391).
 * Forbid non-integers in `:set()` and `datetime.parse()` (gh-10391).
@@ -132,15 +130,13 @@ issues were fixed as part of this activity:
 
 * Fixed a bug when an SQL count statement wasn't tracked by MVCC if it was
   the first in a transaction (gh-10825).
-
-#### Datetime
-
 * Forbid non-integers in SQL's `CAST({}) AS datetime` (gh-10391).
 
 ### Upgrade
 
-* Automatically fix incorrect format of user-defined spaces and empty password
-  for users during upgrade from old schema versions (~1.6.8) (gh-10180).
+* The incorrect format of user-defined spaces and empty password for users
+  during upgrade from old schema versions (~1.6.8) are fixed automatically now
+  (gh-10180).
 
 ### Build
                 
```

</details>

Totktonada commented:
@sergepetrenko Thanks for the update! The patch is OK for me.
